includes/class-wc-stripe-order-handler.php

```diff
@@ -107,22 +107,19 @@ public function process_redirect_payment( $order_id, $retry = true, $previous_er
 			}
 
 			// Prep source object.
-			$source_object           = new stdClass();
-			$source_object->token_id = '';
-			$source_object->customer = $this->get_stripe_customer_id( $order );
-			$source_object->source   = $source_info->id;
-			$source_object->status   = 'chargeable';
+			$prepared_source         = $this->prepare_order_source( $order );
+			$prepared_source->status = 'chargeable';
 
 			/*
 			 * If we're doing a retry and source is chargeable, we need to pass
 			 * a different idempotency key and retry for success.
 			 */
-			if ( $this->need_update_idempotency_key( $source_object, $previous_error ) ) {
+			if ( $this->need_update_idempotency_key( $prepared_source, $previous_error ) ) {
 				add_filter( 'wc_stripe_idempotency_key', [ $this, 'change_idempotency_key' ], 10, 2 );
 			}
 
 			// Make the request.
-			$response = WC_Stripe_API::request( $this->generate_payment_request( $order, $source_object ), 'charges', 'POST', true );
+			$response = WC_Stripe_API::request( $this->generate_payment_request( $order, $prepared_source ), 'charges', 'POST', true );
```

Comment:
I'm really not sure how to trigger this code, but I went with a change that seemed to work for the webhooks.

What I tried to trigger this:

- 3DS payment from the checkout page
- iDeal payment
- 3DS payment from the "pay for order" page (accessible from the **Customer payment page** link in the order details screen).

Reply:
I haven't found any other way to trigger this code. Since the code is too old I was thinking it was something that we forgot to remove. Thanks for leaving instructions!